### ISSUE TITLE
[ROI actions] bug when unchecking a ROI actions

### ISSUE THREAD

**Issue by PiRK** [silx-kit/silx#2655]
When I run the plotInteractiveImageROI.py example, if I select a ROI action (for instance rectangle ROI) then I uncheck it, I get the following error message :

```
Traceback (most recent call last):
  File "C:\Users\Pierre\AppData\Roaming\Python\Python36\site-packages\silx\utils\weakref.py", line 178, in __call__
    return fn(*args, **kwargs)
  File "C:\Users\Pierre\AppData\Roaming\Python\Python36\site-packages\silx\gui\plot\tools\roi.py", line 176, in _modeActionTriggered
    action.setChecked(True)
AttributeError: 'NoneType' object has no attribute 'setChecked'
```
Until I select another ROI action, the previous one remains enabled.

**Comment by PiRK** [comment]
The corresponding code in silx\gui\plot\tools\roi.py line 176 confuses me:
```
        if checked:
            self.start(roiClass)
        else:  # Keep action checked
            action = self.sender()
            action.setChecked(True)
```

What is the recommended way to exit ROI drawing mode?

**Comment by t20100** [comment]
I cannot reproduce the error message.
The code you mention makes sure the action is not unchecked when clicked while checked.

The way to exit ROI draw mode is to select another interaction mode (like zoom or pan mode).
It is the same for the profile tool.

How did you get this error?
It should not happen unless you call the private method directly... but we can add a `action is not None` check to be safe

**Comment by PiRK** [comment]
I got this error by trying to click a ROI action while checked.

The easiest way to reproduce it is to run _examples/plotInteractiveImageROI.py_, and click the "add rectangle ROI" action twice.

When I uncheck a ROI action, the action stays activated but the button is unchecked and I get this error.

**Comment by PiRK** [comment]
Windows 10, silx 0.11, PySide2, backend = gl, python 3.6

**Comment by vasole** [comment]
Have you tried current 0.11 branch? I do not reproduce it with windows 7, silx 0.11, PyQt5, python 3.6

**Comment by PiRK** [comment]
The patch #2656 prevents the error. The remaining issue is that the action remains active although the button is unchecked.

Looks like a Pyside2 bug, then.

**Comment by PiRK** [comment]
I would guess it could be related to the wrapping of the slot in `functools.partial`.
https://srinikom.github.io/pyside-bz-archive/344.html

**Comment by t20100** [comment]
That should be it indeed.

**Comment by t20100** [comment]
I'm making a fix to use a `QActionGroup`.

**Comment by t20100** [comment]
Closed by #2659 on 0.11 branch, but that will be put on master as well once the release is done.
